### PR TITLE
03 GET /api/categories

### PR DESCRIPTION
**Pull request by LukeWatson24**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  if (db.end) db.end();
+});
+
+describe("GET /api/categories", () => {
```
Comment (mezzDavies) [inline __tests__/app.test.js:27]: Looks good for the endpoint but we could also have a separate test and functionality for general 404 error handling at this point too. 😄

File: controllers/app.controllers.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const { fetchCategories } = require("../models/app.models");
+
+const getCategories = (req, res, next) => {
+  return fetchCategories().then((categories) => {
+    res.status(200).send({ categories });
+  });
+};
```
Comment (mezzDavies) [inline controllers/app.controllers.js:12]: We should have a catch block for this controller 👍

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Looking good but a couple of points in the comments. Please have a look and amend accordingly then resubmit your PR. 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -12,6 +12,12 @@ afterAll(() => {
   if (db.end) db.end();
 });
 
+describe("general 404 errors", () => {
+  test("returns 404 if endpoint does not exist", () => {
+    return request(app).get("/api/test").expect(404);
+  });
+});
+
```
Comment (NRMSMITH) [inline __tests__/app.test.js:26]: Great work. 👍

File: app.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const express = require("express");
+const { getCategories } = require("./controllers/app.controllers");
+
+const app = express();
+
+app.get("/api/categories", getCategories);
+
```
Comment (NRMSMITH) [inline app.js:8]: You should add a route here for your 404 error. _hint hint: look at app.all('/*') and think about a new error handling function_.  (you'll need an error controller)

**Review by NRMSMITH — APPROVED** [review]
I've approved as your code is running - please have a look at the comments as they will help you out as you get further through the repo 😄

**Pull request merged**